ci/infra/openstack/generate-cloud-provider-conf

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+
+umask 0077
+
+cat << EOF > openstack.conf
+[Global]
+auth-url="${OS_AUTH_URL}"
+username="${OS_USERNAME}"
+password="${OS_PASSWORD}"
+tenant-name=
+tenant-id=
+domain-name=
+domain-id=
+region="${OS_REGION_NAME}"
+ca-file=
+[LoadBalancer]
+lb-version=v2
+subnet-id="${OS_PRIVATE_SUBNET_ID}"
+floating-network-id="${OS_PUBLIC_NET_ID}"
+create-monitor=yes
+monitor-delay=1m
+monitor-timeout=30s
+monitor-max-retries=3
+[BlockStorage]
+trust-device-path=false
+bs-version=v2
+ignore-volume-az=true
+EOF
+
+umask 0022
+
+if [ -n "$OS_PROJECT_ID" ]; then
```

Comment:
I think you can avoid this complex logic and using `sed` if you follow this pattern:
```
TENANT_ID=${OS_PROJECT_ID:-${OS_PROJECT_NAME}}
if [[ -z TENANT_ID ]]; then
  echo "[ERROR] Either OS_PROJECT_ID or OS_PROJECT_NAME must be specified"
fi
```

You can then use TENANT_ID in the cat command above

Reply:
Yeah, I was thinking about that, but `tenant-id` or `tenant-name` option have to be used in `openstack.conf` file. In this case `sed` is used for setting and cleaning unnecessary option. Using `sed` is good solution, so I would rather to keep it.

Reply:
I have optimized logic for this script, which is still human readable.